extensions/odoo_theme/static/style.scss

```diff
@@ -432,659 +425,471 @@ header {
-            grid-column-end: toc;
+        .toctree-l1 {
+            margin-bottom: 2rem;
         }
 
-        // Homepage
+        .toctree-l2 > a[href="#"] {
+            margin-top: 0.7rem;
+            display: inline-block;
+        }
 
-        &.o_index {
-            .o_content_fw_banner {
-                margin: 0 -0.75rem $margin-m;
-                padding: $padding-m $padding-s;
-                align-items: center;
-                @include o-gradient();
-                @include media-breakpoint-up(lg) {
-                    margin: -#{$margin-m} -#{$margin-l} $margin-m;
-                    padding: $padding-m $padding-m;
-                }
-                @include media-breakpoint-up(xl) {
-                    margin: -#{$margin-m} -#{$margin-l} $margin-l;
-                    padding: $padding-l $padding-l;
-                }
-            }
-            .row:first-of-type h2.border-top {
-                @include media-breakpoint-up(lg) {
-                    border-top: 0 solid transparent !important;
-                }
-            }
+        a[href="#"] {
+            pointer-events: none;
+            cursor: default;
+            color: $o-gray-color;
+            font-weight: $fw_semibold;
+        }
+    }
+    // Toctree on 1st and 2nd level pages - start
 
-            h5 {
-                font-size: 12px;
-            }
+    // Link to edit pages - start
+    .o_git_link {
+        @include font-size($font-size-secondary);
+        @include o-position($top: 0px, $right: $margin-s);
+        @include media-breakpoint-up(lg) {
+            @include o-position($top: 10px, $right: $margin-s);
+        }
+        i {
+            margin-right: .2rem;
+        }
+    }
+    // Link to edit pages - start
 
-            ul {
-                > li {
-                    position: relative;
-                    padding-bottom: $padding-xs;
-                    display: flex;
-                    align-items: center;
+    article.doc-body {
+        padding-top: $padding-s;
+        position: relative;
+    }
 
-                    &:before {
-                        @include o-inline-icon($i-article, 0 .2rem 0 0);
-                        color: $gray-light;
-                    }
+    // References (ie: addons/mixins, js references) - start
+    dl.py, dl.js, dl.o-definition-list {
+        > dd {
+            border-left: 3px solid $o-gray-border;
+            margin: 0;
+            padding-left: $padding-s;
+            padding-top: $padding-xs;
+        }
+        &.class em.property {
+            font-size: $font-size-secondary;
+            &:first-child {
+                border-bottom: 3px solid $o-gray-border;
+                + [class^=sig] {
+                    padding-left: .3rem;
                 }
             }
-
-            .o_changelog {
-                .row:last-of-type > .border-bottom {
-                    border: none !important;
+            &:not(:first-child) {
+                @extend .o_code;
```

Comment:
Don't extend such complex selectors... actually even the simplest ones can lead to serious performance headaches.
Ideally apply the class directly. If it's not possible better to create a mixin ;)